src/middlewared/middlewared/plugins/disk_/sync.py

```diff
@@ -186,7 +183,16 @@ async def sync_all(self, job):
                     await self.middleware.call('datastore.insert', 'storage.disk', disk)
                     changed = True
 
-                await self.middleware.call('enclosure.sync_disk', disk['disk_identifier'], encs)
+        # now we need to check the enclosure mapping db info for each disk
```

Comment:
Should we better have `sync_disks` method in enclosure plugin which will handle an array of disks in one place?